plugins/reindex/src/main/java/org/elasticsearch/plugin/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -179,10 +132,15 @@ public void onFailure(Throwable e) {
     void onScrollResponse(SearchResponse searchResponse) {
         scroll.set(searchResponse.getScrollId());
         if (searchResponse.getShardFailures() != null && searchResponse.getShardFailures().length > 0) {
-            Collections.addAll(searchFailures, searchResponse.getShardFailures());
+            task.addSearchFailures(searchResponse.getShardFailures());
             startNormalTermination();
             return;
         }
+        long total = searchResponse.getHits().totalHits();
+        if (mainRequest.getSize() >= 0) {
```

Comment:
So what happens when someone issues the request with size=0 ? Should we reject that?